provider/marathon_test.go

```diff
@@ -1152,37 +1201,53 @@ func TestMarathonGetEntryPoints(t *testing.T) {
 }
 
 func TestMarathonGetFrontendRule(t *testing.T) {
-	provider := &Marathon{
-		Domain: "docker.localhost",
-	}
-
 	applications := []struct {
-		application marathon.Application
-		expected    string
+		application             marathon.Application
+		expected                string
+		marathonLBCompatibility bool
 	}{
 		{
 			application: marathon.Application{
 				Labels: &map[string]string{}},
-			expected: "Host:.docker.localhost",
+			marathonLBCompatibility: true,
+			expected:                "Host:.docker.localhost",
 		},
 		{
 			application: marathon.Application{
-				ID:     "test",
-				Labels: &map[string]string{},
+				ID: "test",
+				Labels: &map[string]string{
+					"HAPROXY_0_VHOST": "foo.bar",
+				},
 			},
-			expected: "Host:test.docker.localhost",
+			marathonLBCompatibility: false,
+			expected:                "Host:test.docker.localhost",
 		},
 		{
 			application: marathon.Application{
 				Labels: &map[string]string{
 					"traefik.frontend.rule": "Host:foo.bar",
+					"HAPROXY_0_VHOST":       "notvalid",
+				},
+			},
+			marathonLBCompatibility: true,
+			expected:                "Host:foo.bar",
```

Comment:
if `traefik.frontend.rule` is present there is a `return` before trying `HAPROXY` label.

I prefer this behavior with `traefik` labels superseding other one.